exporters/failure/collector_pagerduty.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+#
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+import logging
+
+import requests
+from attrs import define, field
+
+from failure.collector_base import AbstractFailureCollector, TrackerIssue
+from pelorus.config import env_var_names, env_vars
+from pelorus.config.log import REDACT, log
+from pelorus.errors import FailureProviderAuthenticationError
+from pelorus.timeutil import parse_tz_aware, second_precision
+from pelorus.utils import TokenAuth, set_up_requests_session
+
+DEFAULT_PAGERDUTY_URGENCY = "high"
+DEFAULT_PAGERDUTY_PRIORITY = None  # equivalent to null
+
+_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S%z"
+
+
+@define(kw_only=True)
+class PagerdutyFailureCollector(AbstractFailureCollector):
+    """
+    PagerDuty implementation of a FailureCollector
+    """
+
+    token: str = field(
+        default="",
+        metadata=env_vars(*env_var_names.TOKEN) | log(REDACT),
+        repr=False,
+    )
+
+    tls_verify: bool = field(default=True)
+
+    session: requests.Session = field(factory=requests.Session, init=False)
+
+    incident_urgency: str = field(
+        default=DEFAULT_PAGERDUTY_URGENCY, metadata=env_vars("PAGERDUTY_URGENCY")
+    )
+
+    incident_priority: str = field(
+        default=DEFAULT_PAGERDUTY_PRIORITY, metadata=env_vars("PAGERDUTY_PRIORITY")
+    )
+
+    def __attrs_post_init__(self):
+        # disable .netrc
+        self.session.trust_env = False
+
+        if self.token:
+            set_up_requests_session(
+                self.session,
+                self.tls_verify,
+                auth=TokenAuth(self.token, is_pagerduty=True),
+            )
+
+    def get_incidents(self) -> list[dict]:
+        logging.info("Getting incidents")
+        url = "https://api.pagerduty.com/incidents"
+        headers = {
+            "Accept": "application/vnd.pagerduty+json;version=2",
+        }
+
+        resp = self.session.get(url, headers=headers)
```

Comment:
we could have used this one https://github.com/PagerDuty/pdpyras
I felt is was easier to do it directly

Reply:
FYI.. there is a project preference to try not to use 3rd party libraries if we're not going to heavily use the library across the code base.   Hence why you see basic requests being used in cases like this.